sdk/core/azure-core/CMakeLists.txt

```diff
@@ -58,7 +62,7 @@ add_library (Azure::Core ALIAS ${TARGET_NAME})
 create_code_coverage(core ${TARGET_NAME} "${TARGET_NAME}-test")
 
 target_include_directories(${TARGET_NAME} PUBLIC ${CURL_INCLUDE_DIRS})
-target_link_libraries(${TARGET_NAME} PRIVATE CURL::libcurl)
+target_link_libraries(${TARGET_NAME} PRIVATE CURL::libcurl Threads::Threads)
```

Comment:
`<thread>` is used in storage retry policy as well. Should we link `::Threads` there too, or is it sufficient to rely on azure-core linking it (which storage takes a dependency on)?

Reply:
> Should we link `::Threads` there too, or is it sufficient to rely on azure-core linking it (which storage takes a dependency on)?

I just realized, storage common already links the `::Threads` library.

https://github.com/Azure/azure-sdk-for-cpp/blob/a716e094fe4c17f872181cc755e13843f53ead8d/sdk/storage/azure-storage-common/CMakeLists.txt#L44-L50